internal/generate/olm-catalog/package_manifest.go

```diff
@@ -96,41 +95,81 @@ func (g pkgGenerator) Generate() error {
 // generate either reads an existing package manifest or creates a new
 // manifest and modifies it based on values set in s.
 func (g pkgGenerator) generate() (map[string][]byte, error) {
+	fileName, pkg, err := g.buildPackageManifest()
+	if err != nil {
+		return nil, err
+	}
+
+	g.setChannels(&pkg)
+	g.sortChannelsByName(&pkg)
+
+	if err := validatePackageManifest(&pkg); err != nil {
+		return nil, fmt.Errorf("error validating package manifest %s: %v", pkg.PackageName, err)
+	}
+
+	b, err := yaml.Marshal(pkg)
+	if err != nil {
+		return nil, err
+	}
+	fileMap := map[string][]byte{
+		fileName: b,
+	}
+	return fileMap, nil
```

Comment:
Note that above the code is the same. The logic was just encapsulated to bring an easier understanding.